src/tests/unit/components/app-content-switcher/AppContentSwitcher.spec.jsx

```diff
@@ -0,0 +1,49 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { describe, it, vi, expect } from 'vitest'
+import AppContentSwitcher from '~/components/app-content-switcher/AppContentSwitcher'
+
+const mockOnChange = vi.fn()
+
+const switchOptions = {
+  left: { text: 'Left Option', tooltip: 'Left Tooltip' },
+  right: { text: 'Right Option', tooltip: 'Right Tooltip' }
+}
+
+const defaultProps = {
+  active: false,
+  onChange: mockOnChange,
+  switchOptions,
+  typographyVariant: 'body1',
+  styles: {}
+}
+
+describe('AppContentSwitcher', () => {
+  it('should render with the correct props', () => {
+    render(<AppContentSwitcher {...defaultProps} />)
+    expect(screen.getByText('Left Option')).toBeInTheDocument()
+    expect(screen.getByText('Right Option')).toBeInTheDocument()
+    expect(screen.getByTestId('switch')).toBeInTheDocument()
+  })
+
+  it('should call the onChange function when the switch is clicked', async () => {
+    render(<AppContentSwitcher {...defaultProps} />)
+
+    const switchElement = screen.getByRole('checkbox')
+    await userEvent.click(switchElement)
+    expect(mockOnChange).toHaveBeenCalled()
```

Comment:
Should we count calls?
```suggestion
    expect(mockOnChange).toHaveBeenCalledTimes(1)
```
We can also check the arguments passed to this function via `toHaveBeenCalledWith`